projects/auth0-angular/src/lib/auth.service.spec.ts

```diff
@@ -63,29 +80,40 @@ describe('AuthService', () => {
           provide: Auth0ClientService,
           useValue: auth0Client,
         },
+        {
+          provide: AuthConfigService,
+          useValue: authConfig,
+        },
       ],
     };
 
     TestBed.configureTestingModule(moduleSetup);
-    service = createService();
     authState = TestBed.inject(AuthState);
   });
 
+  afterEach(() => {
+    assignMock.mockClear();
+  });
+
   describe('constructor', () => {
     it('should be created', () => {
+      const service = createService();
```

Comment:
Our service does a lot of things in its constructor, meaning it makes sense to have the tests control the creation. Some tests might be fine to have the service created outside of it, but previously we were creating the service outside of every test, and recreating it inside some specific tests.

Important to know is that services are Singletons, so creating the service in the tests doesnt create a new instance if one was created outside of it.

Therefore, all tests now create the instance of the service themselves.